_adr/90/index.adoc

```diff
@@ -0,0 +1,136 @@
+
+## Alternatives Considered / Rejected
+
+1. Deploying https://www.redhat.com/sysadmin/cert-manager-operator-openshift[cert-manager operator] to the data plane. Rejected to avoid exposing domain registrar credentials and to avoid adding another component that consumes data plane resources and that needs to be monitored.
+
+2. Migrating existing their Kafkas. This change is disruptive to existing customers hence why it was rejected. 
+The Kafka fleet manager will thus ensure that existing Kafkas continue to operate using a shared wildcard certificate on the `*.kafka.bf2.dev` domain i.e the proposed solution will only be applied to new Kafkas.
+
+3. Deploying https://www.redhat.com/sysadmin/cert-manager-operator-openshift[cert-manager operator] in the Control Plane. 
+The Control plane then handles the pushing of the Certificate request CR, watching for the created secrets and storing them securely. 
+Rejected because this is technically complex and has no clear benefit over the propsed solution of using a library.
+It also violates the design intentions of the fleet manager by turning into an Kubernetes operator. 
+Finally, it's another component deployed that has to be monitored and managed in the control plane which forces every one (especially dev environments) to have a running Kubernetes cluster running during development
+
+## Challenges
+// What are the costs/drawbacks of the proposed decision?
+1. The drawback is that the solution can't or won't handle the migration of already existing Kafkas.
+This is based on the fact that the proposal intends to make this change a transparent one to existing Kafka instances. 
+For this, the Kafka fleet manager will ensure that existing Kafkas continue to operate using a shared wildcard certificate on the `\*.kafka.bf2.dev` domain. 
+All standard Kafkas will have a wildcard certificate on the `*.<kafka-id>.kafka.bf2.dev` subdomain where the `kafka-id` is unique for each Kafka.
+All new developer Kafkas (short living instances of 48 hours) will have a wildcard certificate on the `*.trial.kafka.bf2.dev` subdomain, and they will share the same wildcard certificate. This is done so as to keep our usage of the https://Letsencrypt.org/[Letsencrypt] sane and avoid frequently reaching their https://Letsencrypt.org/docs/rate-limits/[limits] due to spikes in creation of developer kafkas.
+2. Reaching the various advertised https://Letsencrypt.org/docs/rate-limits/[Letsencrypt limits] is also a challenge.
+However, the various limits once reached can be overridden via a request as described in https://Letsencrypt.org/docs/rate-limits/#a-id-overrides-a-overrides[Letsencrypt limits overrides]. The usage of the https://Letsencrypt.org/[Letsencrypt] is kept sane by sharing an auto managed wildcard certificate for all new developer Kafkas.
+
+## Dependencies
+// What are the knock-on effects if this decision is accepted?
+
+Depedency on Letsencyrpt and its limits 
```

Comment:
I would expect part of the limits of Letsencyrpt can be solved with some enterprise/paid plan.
Am I correct? If so I think we can mention that limits are tied to the current tied that we use

Reply:
No, Letsencrypt doesn't have the notion of enterprise plan as far as I am aware. However, they've the option to request limits override which has been mentioned in this ADR. The process needs to be followed and they (letsencrypt) will get intouch if further clarification is needed.

Other services like zerossl have the notion of plan - and they charge a fee for this.